src/meta/src/manager/catalog/utils.rs

```diff
@@ -0,0 +1,364 @@
+
+    match &mut stmt {
+        Statement::CreateTable {
+            query: Some(query), ..
+        }
+        | Statement::CreateView { query, .. }
+        | Statement::Query(query) // Used by view, actually we store a query as the definition of view.
+        | Statement::CreateSink {
+            stmt:
+                CreateSinkStatement {
+                    sink_from: CreateSink::AsQuery(query),
+                    ..
+                },
+        } => {
+            QueryRewriter::rewrite_query(query, from, to);
+        }
+        Statement::CreateIndex { table_name, .. }
+        | Statement::CreateSink {
+            stmt:
+                CreateSinkStatement {
+                    sink_from: CreateSink::From(table_name),
+                    ..
+                },
+        } => replace_table_name(table_name, to),
+        _ => unreachable!(),
+    };
+    stmt.to_string()
+}
+
+/// Replace the last ident in the `table_name` with the given name, the object name is ensured to be
+/// non-empty. e.g. `schema.table` or `database.schema.table`.
+fn replace_table_name(table_name: &mut ObjectName, to: &str) {
+    let idx = table_name.0.len() - 1;
+    table_name.0[idx] = Ident::new_unchecked(to);
+}
+
+/// `QueryRewriter` is a visitor that updates all references of relation named `from` to `to` in the
+/// given query, which is the part of create statement of `relation`.
+struct QueryRewriter<'a> {
+    from: &'a str,
+    to: &'a str,
+}
+
+impl QueryRewriter<'_> {
+    fn rewrite_query(query: &mut Query, from: &str, to: &str) {
+        let rewriter = QueryRewriter { from, to };
+        rewriter.visit_query(query)
+    }
+
+    /// Visit the query and update all references of relation named `from` to `to`.
+    fn visit_query(&self, query: &mut Query) {
+        if let Some(with) = &mut query.with {
+            for cte_table in &mut with.cte_tables {
+                self.visit_query(&mut cte_table.query);
+            }
+        }
+        self.visit_set_expr(&mut query.body);
+        for expr in &mut query.order_by {
+            self.visit_expr(&mut expr.expr);
+        }
+    }
+
+    /// Visit table factor and update all references of relation named `from` to `to`.
+    /// Rewrite idents(i.e. `schema.table`, `table`) that contains the old name in the
+    /// following pattern:
+    /// 1. `FROM a` to `FROM new_a AS a`
+    /// 2. `FROM a AS b` to `FROM new_a AS b`
+    ///
+    /// So that we DON'T have to:
+    /// 1. rewrite the select and expr part like `schema.table.column`, `table.column`,
+    /// `alias.column` etc.
+    /// 2. handle the case that the old name is used as alias.
+    /// 3. handle the case that the new name is used as alias.
+    fn visit_table_factor(&self, table_factor: &mut TableFactor) {
+        match table_factor {
+            TableFactor::Table { name, alias, .. } => {
+                let idx = name.0.len() - 1;
+                if name.0[idx].real_value() == self.from {
```

Comment:
Is it possible that this refers to a table in a different (pg) schema? 🤔

Reply:
Yes, it's possible and renaming works in current implementation in rw.